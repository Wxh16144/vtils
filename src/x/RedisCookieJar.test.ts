import { RedisCookieJar } from './RedisCookieJar'

jest.mock('ioredis', () => {
  const ioredis = require('ioredis-mock')
  Object.assign(ioredis, jest.requireActual('ioredis'))
  return ioredis
})

describe('RedisCookieJar', () => {
  test('表现正常', async () => {
    const { default: Redis } = await import('ioredis')
    const redis = new Redis()
    const cookieJar = new RedisCookieJar({ key: 'test', redis })
    await cookieJar.setCookie('sid=12345', 'https://baidu.com')
    expect(await redis.keys('*')).toMatchSnapshot()
    expect(await redis.get((await redis.keys('*'))[0])).toMatchSnapshot()
  })
})
