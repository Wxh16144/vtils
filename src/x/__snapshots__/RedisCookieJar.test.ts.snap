// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RedisCookieJar 表现正常 1`] = `
Array [
  "@@@@COOKIE@@@@test",
]
`;

exports[`RedisCookieJar 表现正常 2`] = `"{\\"baidu.com\\":{\\"/\\":{\\"sid\\":{\\"key\\":\\"sid\\",\\"value\\":\\"12345\\",\\"domain\\":\\"baidu.com\\",\\"path\\":\\"/\\",\\"hostOnly\\":true,\\"pathIsDefault\\":true,\\"creation\\":\\"2020-10-31T16:00:00.000Z\\",\\"lastAccessed\\":\\"2020-10-31T16:00:00.000Z\\"}}}}"`;
